server/authorization.go

```diff
@@ -19,23 +19,28 @@ func (ae *authError) Error() string {
 	return string(errorString)
 }
 
-var oAuthMessage string = "[Click here to link your Microsoft account.](%s/plugins/" + manifest.Id + "/oauth2/connect?channelID=%s)"
+func (p *Plugin) getOauthMessage() (string, error) {
+	pluginOauthURL, err := p.getPluginOauthURL()
+	if err != nil {
+		return "", err
+	}
+
+	return "[Click here to link your Microsoft account.](" + pluginOauthURL + "/connect?channelID=%s)", nil
```

Comment:
It doesn't make sense to return this string if you're always going to `oauthMsg := fmt.Sprintf(oAuthMessage, channelID)` when you use the return value.

Plus `oauthMsg := fmt.Sprintf(oAuthMessage, channelID)` is hard to understand what is evening happening without referencing this function.

You should combine them into one function.

Reply:
@enzowritescode Fixed. Please re-review